src/diffusers/pipelines/pixart_alpha/pipeline_pixart_sigma.py

```diff
@@ -859,7 +866,7 @@ def __call__(
                     progress_bar.update()
                     if callback is not None and i % callback_steps == 0:
                         step_idx = i // getattr(self.scheduler, "order", 1)
-                        callback(step_idx, t, latents)
+                        callback(self, step_idx, t, latents) #Not 100% sure if this will break anything. Callback documentation would need to be updated to to reflect the added input
```

Comment:
actually can you help refactor it to use the new callback API with `callback_on_step_end` and `callback_on_step_end_tensor_inputs` (see an example https://github.com/huggingface/diffusers/blob/e7b9a0762bd015372ea56767809bc2132f6b4dad/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L779)